Framework/PythonInterface/plugins/algorithms/LoadElementalAnalysisData.py

```diff
@@ -0,0 +1,179 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+
+import glob
+import os
+import numpy as np
+from mantid import config
+from mantid.api import AlgorithmFactory, AnalysisDataService, PythonAlgorithm, TextAxis, WorkspaceGroup, WorkspaceGroupProperty
+from mantid.kernel import Direction, IntBoundedValidator
+from mantid.simpleapi import ConvertToHistogram, ConvertToPointData, DeleteWorkspace, LoadAscii, WorkspaceFactory
+
+
+type_keys = {"10": "Prompt", "20": "Delayed", "99": "Total"}
+spectrum_index = {"Delayed": 1, "Prompt": 2, "Total": 3}
+num_files_per_detector = 3
+run_number_length = 5
+
+
+class LoadElementalAnalysisData(PythonAlgorithm):
+    def category(self):
+        """
+        Returns category
+        """
+        return "DataHandling"
+
+    def PyInit(self):
+        self.declareProperty(name='Run', defaultValue=0, validator=IntBoundedValidator(lower=0),
+                             doc='Run number to load')
+
+        self.declareProperty(WorkspaceGroupProperty(name='GroupWorkspace', defaultValue='',
+                                                    direction=Direction.Output),
+                             doc='Output group workspace for run')
+
+    def validateInputs(self):
+        issues = dict()
+        checkRun = self.search_user_dirs()
+        if not checkRun:
+            issues['Run'] = "Cannot find files for run " + self.getPropertyValue("Run")
+        return issues
+
+    def PyExec(self):
+        run = self.getPropertyValue("Run")
+        to_load = self.search_user_dirs()
+        workspaces = {
+            path: self.get_filename(path, run)
+            for path in to_load if self.get_filename(path, run) is not None
+        }
+        unique_workspaces = {}
+        for path, workspace in workspaces.items():
+            if workspace not in unique_workspaces.values():
+                unique_workspaces[path] = workspace
+        workspaces = unique_workspaces
+
+        self.format_workspace(workspaces)
+        self.merge_workspaces(workspaces.values())
+
+    def pad_run(self):
+        """ Pads run number: i.e. 123 -> 00123; 2695 -> 02695 """
+        return str(self.getProperty("Run").value).zfill(run_number_length)
+
+    def search_user_dirs(self):
+        """ Finds all files for the run number provided """
```

Comment:
Just a question: Is the intention here to find all possible files that match the format across all of the search paths? This is slightly inconsistent with the rest of the loaders that pick the first thing they find. For a setup that adds a lot of search paths, IDAaaS can be setup this way with 100s of directories, then this could end up doing a lot of file accessing.

Reply:
Yes that is the intention. The algorithm has been developed from 
https://github.com/mantidproject/mantid/blob/b4d0543e9c6bf0c97abf90b9ca607983864cf3d5/scripts/Muon/GUI/ElementalAnalysis/LoadWidget/load_utils.py
which has this functionality. Its how the current Elemental Analysis GUI loads runs. I suspect the reason for this is that each spectra for each detector for the run is saved in a separate file so you'd need to make sure all files with the same run number are loaded to get the full results. We certainly don't want the algorithm to return only the first thing they find.

I'm not sure how this issue of searching multiple directories in IDAaaS has been dealt with for Elemental Analysis in the past - @AnthonyLim23 do you know?

Reply:
Thanks for pointing out the old functionality. My apologies for not being more clear about my meaning on the search. I was more thinking whether we would expect the files to be scattered across several directories or they should be expected to be in a single directory together so you could stop after the first directory that found something.

Reply:
I think for a given run they should all be in the same directory. We have not implemented anything for IDAaas specifically.

Reply:
Should we make that a requirement then and break out of the directory search as soon as files are found in one of the directories - it makes it more consistent with the rest of the file searching in the framework?

Reply:
@sf1919 could you double check with Aidy that this would be ok?

Reply:
Confirmed that break was the correct approach.

Implemented in commit fd81a40